### PR TITLE
feat: add support for client TLS certs

### PR DESCRIPTION
**Pull request by christian-roggia**
## Related issue

https://github.com/ory/oathkeeper/pull/744

### CODE REVIEW

File: helm/charts/oathkeeper/templates/deployment-controller.yaml [hunk 1]
```diff
@@ -42,16 +42,21 @@ spec:
...
           configMap:
-            name: {{ include "oathkeeper.fullname" . }}-rules
+            name: {{ include "oathkeeper.name" . }}-rules
         - name: {{ include "oathkeeper.name" . }}-secrets-volume
           secret:
             secretName: {{ empty .Values.secret.name | ternary (include "oathkeeper.fullname" .) (.Values.secret.name) }}
+        {{- if .Values.certs.enabled }}
+        - name: {{ include "oathkeeper.name" . }}-certs
+          secret:
+            secretName: {{- if .Values.certs.existingSecret }}
```
Comment (Demonsthere) [inline helm/charts/oathkeeper/templates/deployment-controller.yaml:58]:
```suggestion
            secretName: {{- .Values.certs.existingSecret }}
```

File: helm/charts/oathkeeper/templates/deployment-controller.yaml [hunk 2]
```diff
@@ -42,16 +42,21 @@ spec:
         - name: {{ include "oathkeeper.name" . }}-config-volume
           configMap:
             {{- if .Values.demo }}
-            name: {{ include "oathkeeper.fullname" . }}-config-demo
+            name: {{ include "oathkeeper.name" . }}-config-demo
```
Comment (Demonsthere) [inline helm/charts/oathkeeper/templates/deployment-controller.yaml:45]: Please do not change this, and use `fullname`. This allows installation of multiple instances without resource overlapping.

File: helm/charts/oathkeeper/templates/deployment-controller.yaml [hunk 3]
```diff
@@ -42,16 +42,21 @@ spec:
...
             {{- end }}
         - name: {{ include "oathkeeper.name" . }}-rules-volume
           configMap:
-            name: {{ include "oathkeeper.fullname" . }}-rules
+            name: {{ include "oathkeeper.name" . }}-rules
         - name: {{ include "oathkeeper.name" . }}-secrets-volume
           secret:
             secretName: {{ empty .Values.secret.name | ternary (include "oathkeeper.fullname" .) (.Values.secret.name) }}
+        {{- if .Values.certs.enabled }}
+        - name: {{ include "oathkeeper.name" . }}-certs
```
Comment (Demonsthere) [inline helm/charts/oathkeeper/templates/deployment-controller.yaml:56]:
```suggestion
        - name: {{ include "oathkeeper.fullname" . }}-certs
```

File: helm/charts/oathkeeper/templates/deployment-controller.yaml [hunk 4]
```diff
@@ -92,6 +97,11 @@ spec:
             - name: {{ include "oathkeeper.name" . }}-secrets-volume
               mountPath: {{ .Values.secret.mountPath }}
               readOnly: true
+            {{- if .Values.certs.enabled }}
+            - name: {{ include "oathkeeper.name" . }}-certs
```
Comment (Demonsthere) [inline helm/charts/oathkeeper/templates/deployment-controller.yaml:101]:
```suggestion
            - name: {{ include "oathkeeper.fullname" . }}-certs
```

**Review by Demonsthere — CHANGES_REQUESTED** [review]
Hello there! Thanks for the PR :) 
Got a few suggestions

**Review by aeneasr — CHANGES_REQUESTED** [review]
I think it would make sense to use the extraVolume pattern we have in Ory Kratos instead: https://github.com/ory/k8s/blob/master/helm/charts/kratos/values.yaml#L167-L179 :)

**Comment by christian-roggia** [comment]
I can work on introducing that pattern

**Comment by aeneasr** [comment]
As per https://github.com/ory/oathkeeper/pull/744#issuecomment-878235782 marking this as draft

**Comment by tobbbles** [comment]
There's a hard and fast addition of extraVolume's at #330 -  I found myself to need it to get TLS certs into the pod :)

**Comment by Demonsthere** [comment]
I'd say lets deprecate this PR in favour of #330. @christian-roggia are you ok with this?

**Comment by christian-roggia** [comment]
Yes.

**Pull request closed without merging**